rootfs/etc/s6/mariadb/run

```diff
@@ -29,18 +29,17 @@ then
   echo "DROP DATABASE IF EXISTS test;" >> ${INIT_FILE}
   echo "FLUSH PRIVILEGES;" >> ${INIT_FILE}
 
-  if [[ -n "${MARIADB_USERNAME}" && -n "${MARIADB_DATABASE}" ]]
+  MARIADB_DATABASE=${MARIADB_DATABASE:-$MARIADB_USERNAME}
+  if [ ! -z "${MARIADB_DATABASE}" ]
```

Comment:
I know now, but at the time the wacky documentation on http://tldp.org/LDP/abs/html/comparison-ops.html and the people on stackoverflow quoting it said that `-n` tests for "string is not NULL".
However they seem to be wrong, as running `help test` shows `-n` "True if string is not EMPTY."
(I needed to check for EMPTY, not for NULL, so I went with `! -z`)

I guess we can also drop those ugly acolades {} inside the double quotes.
They seem to have no purpose there.
And drop the `-n` or the `! -z` also, as it is irrelevant.
And we'll end up with :  if [ "$MARIADB_DATABASE" ]
Or with double brackets (which have support for no double quotes):  if [[ $MARIADB_DATABASE ]]